src/main/java/app/studnicki/ox/Game.java

```diff
@@ -0,0 +1,172 @@
+package app.studnicki.ox;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+
+/**
+ * Class for playing the ox game.
+ * It provides all of the game logic when other variables
+ * like players, rules, ui are given in constructor.
+ * Maximum of 3 rounds can be played.
+ *
+ * @author Aleksander Studnicki
+ */
+public class Game implements PropertyChangeListener {
+  final Player player1;
+  final Player player2;
+  Queue<Player> playersQueue;
+  BoardChecker boardChecker;
+  int dimension;
+  UserInterface userInterface;
+
+  private final List<Round> rounds = new LinkedList<>();
+  private Round tail;
+
+  private Game(Player player1, Player player2, int winningRule,
+               int dimension, UserInterface userInterface) {
+    this.player1 = player1;
+    this.player2 = player2;
+    playersQueue = new LinkedList<>(List.of(player1, player2));
```

Comment:
This should be done outside of this class, in ctor call, like so: `new Game(new LinkedList<Player>(List.of(p1,p2)...`

Reply:
See comment above about player1 and player2.

Reply:
See comment about player1, player2 issue.